MCE Console Fixes

eemurphy opened the pull request:
Signed-off-by: Erin Murphy <erinmurp@redhat.com>

Issue: https://github.com/stolostron/backlog/issues/26671

Jira: https://issues.redhat.com/browse/ACM-1983

JakobGray commented on controllers/backplaneconfig_controller.go:
```diff
@@ -328,14 +328,24 @@ func (r *MultiClusterEngineReconciler) SetupWithManager(mgr ctrl.Manager) error
+						tmpreq := reconcile.Request{
+							NamespacedName: types.NamespacedName{
+								Name:      mce.GetName(),
+								Namespace: mce.GetNamespace(),
```
MCE doesn't have a namespace

JakobGray commented on pkg/status/console_test.go:
```diff
@@ -194,54 +201,3 @@ func Test_consoleGetKind(t *testing.T) {
 	}
 }
-
-func Test_statusUnknown(t *testing.T) {
```
Why is this test removed?

eemurphy replied:
We don't want a statusUnknown error, it would error out to statusUnkown usually before adding the Console disabled status. I believe we want to have it have the console disabled status in replace of statusUnknown. I noticed that the error I was sending to statusUnkown method on `apierrors.IsNotFound(err)` would be true always when console was not found, and that is why we are adding this custom status instead. So I deleted the check that sent to unknownStatus, therefore we don't need to test it anymore

JakobGray replied:
Gotcha. I thought there was a statusUnknown function missing a test.

JakobGray reviewed (changes requested):
Tested on OCP 4.12 with console disabled. Could not delete MCE. Error in logs:
```
1.6687089606487985e+09	INFO	Failed to find console: cluster	{"controller": "multiclusterengine", "controllerGroup": "multicluster.openshift.io", "controllerKind": "MultiClusterEngine", "multiClusterEngine": {"name":"dummy-multiclusterengine"}, "namespace": "", "name": "dummy-multiclusterengine", "reconcileID": "a3ca5f53-95b5-4944-a161-50a425314576"}
1.6687089606488295e+09	INFO	Error ensuring plugin is removed from console resource	{"controller": "multiclusterengine", "controllerGroup": "multicluster.openshift.io", "controllerKind": "MultiClusterEngine", "multiClusterEngine": {"name":"dummy-multiclusterengine"}, "namespace": "", "name": "dummy-multiclusterengine", "reconcileID": "a3ca5f53-95b5-4944-a161-50a425314576"}
```

eemurphy commented:
> Tested on OCP 4.12 with console disabled. Could not delete MCE. Error in logs:
> 
> ```
> 1.6687089606487985e+09	INFO	Failed to find console: cluster	{"controller": "multiclusterengine", "controllerGroup": "multicluster.openshift.io", "controllerKind": "MultiClusterEngine", "multiClusterEngine": {"name":"dummy-multiclusterengine"}, "namespace": "", "name": "dummy-multiclusterengine", "reconcileID": "a3ca5f53-95b5-4944-a161-50a425314576"}
> 1.6687089606488295e+09	INFO	Error ensuring plugin is removed from console resource	{"controller": "multiclusterengine", "controllerGroup": "multicluster.openshift.io", "controllerKind": "MultiClusterEngine", "multiClusterEngine": {"name":"dummy-multiclusterengine"}, "namespace": "", "name": "dummy-multiclusterengine", "reconcileID": "a3ca5f53-95b5-4944-a161-50a425314576"}
> ```

Just added a commit that should address this, just deleted this extra check from my previous commit by accident

JakobGray commented on controllers/toggle_components.go:
```diff
@@ -90,21 +90,23 @@ func (r *MultiClusterEngineReconciler) ensureNoConsoleMCE(ctx context.Context, b
-		if err != nil {
-			log.Error(err, fmt.Sprintf("Failed to delete Console MCE template: %s", template.GetName()))
-			return result, err
+		if ocpConsole && !strings.Contains(strings.ToLower(template.GetAPIVersion()), "console") {
```
This causes resources in the console.openshift.io apigroup to not be cleaned up when console is disabled

JakobGray reviewed (changes requested):
Also tested this image with a 4.11 OCP cluster (console available). Disabling the console component does not remove all console resources. I indicated where and why this happens.

JakobGray commented:
/override ci/prow/sonar-pre-submit
